overlord/snapstate/handlers.go

```diff
@@ -4078,34 +4078,42 @@ func (m *SnapManager) undoMigrateSnapHome(t *state.Task, tomb *tomb.Tomb) error
 	return writeMigrationStatus(t, snapst, snapsup)
 }
 
-func (m *SnapManager) doEnforceValidationSets(t *state.Task, _ *tomb.Tomb) error {
-	st := t.State()
-	st.Lock()
-	defer st.Unlock()
-
-	var userID int
-	if err := t.Get("userID", &userID); err != nil {
-		return err
-	}
-
+func (m *SnapManager) decodeValidationSets(t *state.Task) (map[string]*asserts.ValidationSet, error) {
 	encodedAsserts := make(map[string][]byte)
 	if err := t.Get("validation-sets", &encodedAsserts); err != nil {
-		return err
+		return nil, err
 	}
 
 	decodedAsserts := make(map[string]*asserts.ValidationSet, len(encodedAsserts))
 	for vsStr, encAssert := range encodedAsserts {
 		decAssert, err := asserts.Decode(encAssert)
 		if err != nil {
-			return err
+			return nil, err
 		}
 
 		vsAssert, ok := decAssert.(*asserts.ValidationSet)
 		if !ok {
-			return errors.New("expected encoded assertion to be of type ValidationSet")
+			return nil, errors.New("expected encoded assertion to be of type ValidationSet")
 		}
 		decodedAsserts[vsStr] = vsAssert
 	}
+	return decodedAsserts, nil
+}
+
+func (m *SnapManager) doEnforceValidationSets(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	// 'local' determines which enforcement function to invoke. If local is set
+	// to true, then we should call EnforceLocalValidationSets, which does not
+	// fetch any assertions or their pre-requisites. If local is set to false, then
+	// we can call EnforceValidationSets, which may contact the store for any additional
+	// assertions.
+	var local bool
+	if err := t.Get("local", &local); err != nil {
```

Comment:
we need to ignore it's missing to support old task

Reply:
I really like this suggestion!